File: Task/Nth/Nim/nth.nim
const Suffix = ["th", "st", "nd", "rd", "th", "th", "th", "th", "th", "th"]

proc nth(n: Natural): string =
  $n & "'" & (if n mod 100 in 11..20: "th" else: Suffix[n mod 10])

for j in countup(0, 1000, 250):
  for i in j..j+24:
    stdout.write nth(i), " "
  echo ""

File: Task/Nth/PascalABC.NET/nth.pas
var suffix := |'th', 'st', 'nd', 'rd'| + |'th'| * 6;

function Nth(n: integer)
  := $'{n}''' + if n mod 100 not in 11..19 then suffix[n mod 10] else 'th';

begin
  ((0..24) + (500..524) + (700..724) + (1000..1024))
    .Select(Nth).Println;
end.
